src/request/sign-transaction/SignTransaction.js

```diff
@@ -1,8 +1,69 @@
 /* global Nimiq */
 /* global TransactionType */
 /* global KeyStore */
-/* global Utf8Tools */
+/* global Identicon */
+
 class SignTransaction { // eslint-disable-line no-unused-vars
+    /**
+     * @param {TransactionRequest} txRequest
+     * @returns {HTMLDivElement}
+     */
+    fillTransactionDetails(txRequest) {
+        /** @type {HTMLDivElement} */
+        const $transaction = (document.querySelector('#transaction-data .transaction'));
+
+        /** @type {HTMLDivElement} */
+        const $senderIdenticon = ($transaction.querySelector('#sender-identicon'));
+        /** @type {HTMLDivElement} */
+        const $recipientIdenticon = ($transaction.querySelector('#recipient-identicon'));
+
+        /** @type {HTMLDivElement} */
+        const $senderLabel = ($transaction.querySelector('#sender-label'));
+        /** @type {HTMLDivElement} */
+        const $recipientLabel = ($transaction.querySelector('#recipient-label'));
+
+        /** @type {HTMLDivElement} */
+        const $senderAddress = ($transaction.querySelector('#sender-address'));
+        /** @type {HTMLDivElement} */
+        const $recipientAddress = ($transaction.querySelector('#recipient-address'));
+
+        /** @type {HTMLDivElement} */
+        const $value = ($transaction.querySelector('#value'));
+        /** @type {HTMLDivElement} */
+        const $fee = ($transaction.querySelector('#fee'));
+        /** @type {HTMLDivElement} */
+        const $message = ($transaction.querySelector('#message'));
+
+        // Set data
+        new Identicon(txRequest.sender, $senderIdenticon); // eslint-disable-line no-new
+        new Identicon(txRequest.recipient, $recipientIdenticon); // eslint-disable-line no-new
+
+        if (txRequest.senderLabel) {
+            $senderLabel.classList.remove('display-none');
+            $senderLabel.textContent = txRequest.senderLabel;
+        }
+        if (txRequest.recipientLabel) {
+            $recipientLabel.classList.remove('display-none');
+            $recipientLabel.textContent = txRequest.recipientLabel;
+        }
+
+        $senderAddress.textContent = txRequest.sender;
+        $recipientAddress.textContent = txRequest.recipient;
+
+        $value.textContent = Nimiq.Policy.satoshisToCoins(txRequest.value).toString();
+        if (txRequest.fee > 0) {
+            $fee.textContent = Nimiq.Policy.satoshisToCoins(txRequest.fee).toString();
+            const $feeSection = $transaction.querySelector('.fee-section');
+            if ($feeSection) $feeSection.classList.remove('display-none');
```

Comment:
why is this check necessary?

Reply:
Because I forgot to typecast the `$feeSection`. Fixed it.